h2o-algos/src/main/java/hex/tree/xgboost/XGBoostScoreTask.java

```
@@ -0,0 +1,205 @@
+
+            float[] weights = data.getWeight();
+
+            if (_output.nclasses() == 1) {
+                double[] dpreds = new double[preds.length];
+                for (int j = 0; j < dpreds.length; ++j)
+                    dpreds[j] = preds[j][0];
+                for (int i = 0; i < cs[0]._len; ++i) {
+                    ncs[0].addNum(dpreds[i]);
+                    ncs[1].addNum(labels[i]);
+                }
+            } else if (_output.nclasses() == 2) {
+                double[] dpreds = new double[preds.length];
+
+                for (int j = 0; j < dpreds.length; ++j)
+                    dpreds[j] = preds[j][0];
+
+                if (weights.length > 0)
+                    for (int j = 0; j < dpreds.length; ++j)
+                        assert weights[j] == 1.0;
+
+                for (int i = 0; i < cs[0]._len; ++i) {
+                    double p = dpreds[i];
+                    ncs[1].addNum(1.0d - p);
+                    ncs[2].addNum(p);
+                    double[] row = new double[]{0, 1 - p, p};
+                    double predLab = hex.genmodel.GenModel.getPrediction(row, _output._priorClassDist, null, Model.defaultThreshold(_output));
+                    ncs[0].addNum(predLab);
+
+                    ncs[3].addNum(labels[i]);
+                }
+            } else {
+                for (int i = 0; i < cs[0]._len; ++i) {
+                    double[] row = new double[ncs.length - 1];
+                    for (int j = 1; j < row.length; ++j) {
+                        double val = preds[i][j - 1];
+                        ncs[j].addNum(val);
+                        row[j] = val;
+                    }
+                    ncs[0].addNum(hex.genmodel.GenModel.getPrediction(row, _output._priorClassDist, null, Model.defaultThreshold(_output)));
+                    ncs[ncs.length - 1].addNum(labels[i]);
+                }
+            }
+        } catch (XGBoostError xgBoostError) {
+            xgBoostError.printStackTrace();
```

Swallowing exception?